controllers/openstackmachine_controller.go

```diff
@@ -448,16 +445,15 @@ func (r *OpenStackMachineReconciler) getOrCreate(logger logr.Logger, cluster *cl
 		logger.Info("Machine not exist, Creating Machine", "Machine", openStackMachine.Name)
 		instanceSpec, err := machineToInstanceSpec(openStackCluster, machine, openStackMachine, userData)
 		if err != nil {
-			err = errors.Errorf("machine spec is invalid: %v", err)
-			handleUpdateMachineError(logger, openStackMachine, err)
+			err = fmt.Errorf("machine spec is invalid: %w", err)
```

Comment:
Oh yes, that's right.

Reply:
I checked again.
The `machineToInstanceSpec` will only return an error, if `openstackmachine == nil`.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/505b5d27c834e5929e5f076cdfe47ff6b97de381/controllers/openstackmachine_controller.go#L468-L470

The OpenStackMachine is initialized here:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/505b5d27c834e5929e5f076cdfe47ff6b97de381/controllers/openstackmachine_controller.go#L82-L83

So whenever this fails I think it would be good to make it a transient error and retry. Therefore I'll suggest to keep the implementation proposed in this PR, and not set failureMessage and failureReason.

Reply:
@seanschneeweiss Conceptually an error here would be non-transient. However, in practise this method can't actually return an error. We should remove the nil check[1] and error return value until it can actually return an error.  That will prompt us to re-evaluate this when it actually means something.

This change is fine, though. The code just needs a cleanup.

[1] We don't need the nil check here because openStackMachine can't be nil and we'd have panicked long before if it was. If we want proper nil pointer safety we'd need to change language.

Reply:
nil check and error return value removed. :+1: Thank you for that suggestion.